src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_SIZE = 10;
+    private Object[] values;
+    private int size;
+    private int arraySize = DEFAULT_SIZE;
+    private T removedObject;
 
+    public ArrayList() {
+        values = new Object[DEFAULT_SIZE];
     }
 
-    @Override
-    public void add(T value, int index) {
+    public ArrayList(int initialCapacity) {
+        values = new Object[initialCapacity];
+    }
 
+    @Override
+    public int size() {
+        return size;
     }
 
     @Override
-    public void addAll(List<T> list) {
+    public void add(T object) {
+        resize();
+        values[size] = object;
+        size++;
+    }
 
+    @Override
+    public void add(T object, int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bound!");
+        }
+        resize();
+        System.arraycopy(values, index, values, index + 1, size - index);
+        values[index] = object;
+        ++size;
     }
 
     @Override
-    public T get(int index) {
-        return null;
+    public void addAll(List<T> list) {
+        if (list.isEmpty() || list.size() == 0) {
+            return;
+        }
+        while (list.size() > values.length) {
+            resize();
+        }
+        int j = 0;
+        for (int i = size;j < list.size();i++) {
+            values[i] = list.get(j);
+            j++;
+            size++;
+        }
```

Comment:
```suggestion
        for (int i = 0; i < list.size(); i++) {
            add(list.get(i));
        }
```

Reply:
isEmpty() and size()==0 are the same thing
we don't need to handle this specifically, it it's the case the loop won't run even once. Resize is called inside add()